robot_calibration/include/robot_calibration/optimize_parameters.h

```diff
@@ -0,0 +1,71 @@
+#ifndef ROBOT_CALIBRATION_OPTIMIZE_PARAMETERS_H
+#define ROBOT_CALIBRATION_OPTIMIZE_PARAMETERS_H
+
+
+/**
+* @brief      Verifies the required parameters have been set and, if not, set default values.
+*
+* @param[in]  node  node handle reference
+*
+* @return     bool indicating whether the parameters are all set.
+*/
+bool check_parameters(ros::NodeHandle& node);
```

Comment:
I really do like this pattern of refactoring the work into functions.  That makes it easier to test and use.